segmentstore/server/src/main/java/io/pravega/segmentstore/server/store/StreamSegmentContainerRegistry.java

```diff
@@ -168,6 +186,14 @@ private void unregisterContainer(ContainerWithHandle containerWithHandle) {
         // Notify the handle that the container is now in a Stopped state.
         containerWithHandle.handle.notifyContainerStopped();
         log.info("Unregistered SegmentContainer {}.", containerWithHandle.handle.getContainerId());
+
+        containerWithHandle.shutdownNotifier.complete(null);
+    }
+
+    private static boolean isShutdown(Service.State state) {
+        return state == Service.State.FAILED
+                || state == Service.State.STOPPING
+                || state == Service.State.TERMINATED;
```

Comment:
This is not really a question about the changes here, but I was looking into understanding why we need a `TERMINATED` state. Is this a transient state until we get rid of the container instance? Also, to be consistent, shouldn't we call it `STOPPER` as the preceding state is `STOPPING`?